storage/redis_adapter.go

```diff
@@ -29,11 +29,11 @@ type messagePipeliner struct {
 	errCh         chan error
 }
 
-func newMessagePipeliner(bufferSize int, redisClient *r.Client, errCh chan error) *messagePipeliner {
+func newMessagePipeliner(bufferSize int, redisClient *r.Client, timeoutSeconds int, errCh chan error) *messagePipeliner {
```

Comment:
yea, fair enough. I was suggesting converting it to a `time.Duration` at the config struct, and passing `time.Duration`s down the stack from there. thoughts?